primitives/consensus/beefy/src/lib.rs

```diff
@@ -303,15 +322,14 @@ impl OpaqueKeyOwnershipProof {
 }
 
 sp_api::decl_runtime_apis! {
-	/// API necessary for BEEFY voters.
-	#[api_version(2)]
-	pub trait BeefyApi
+	/// API necessary for BEEFY voters with only ECDSA key.
+	pub trait BeefyApi<AuthorityId> where AuthorityId : Encode + Decode
```

Comment:
Is this some intermediary step before supporting full API using other than ECDSA?

With this new code, the BEEFY runtime API is generic on `AuthorityId`, but the equivocation functions are using hardcoded `ecdsa_crypto::AuthorityId` which is inconsistent/incorrect when generic `AuthorityId` is other than ECDSA.

What is keeping us from making full api generic and consistent?

Reply:
Resolved according to our discussion.